### PR TITLE
Add migration delete command

### PR DESCRIPTION
**Pull request by Drab-Hounet**
* add command in order to delete migration according to the id of the migration

### CODE REVIEW

File: tests/unit/commands/migration/test_migration.py [hunk 1]
```diff
@@ -71,6 +73,54 @@ def test_do_list_return_no_migration_message_when_there_is_no_migration(self, mo
...
+        m.do_delete(args)
+
+        # then
+        mock_migration_table.assert_called_with([migration])
+
+    @patch('hammr.utils.migration_utils.migration_table')
+    @patch('uforge.application.Api._Users._Migrations.Delete')
+    @patch('uforge.application.Api._Users._Migrations.Get')
+    @patch('ussclicore.utils.generics_utils.query_yes_no')
+    def test____(self, mock_query_yes_no, mock_api_get, mock_api_delete, mock_migration_table):
```
Comment (cinlloc) [inline tests/unit/commands/migration/test_migration.py]: Naming ?

File: tests/unit/commands/migration/test_migration.py [hunk 2]
```diff
@@ -29,7 +29,8 @@
 class TestMigration(unittest.TestCase):
```
Comment (cinlloc) [inline tests/unit/commands/migration/test_migration.py]: Don't we need to verify we call self.api.Users(self.login).Migrations(doArgs.id).Delete() at some point?

File: documentation/en/source/pages/command-line/migration.rst [hunk 3]
```diff
@@ -18,4 +18,14 @@ Sub Commands
...
 
-Displays all the migrations for the user.
\ No newline at end of file
+Displays all the migrations for the user.
+
+
+``delete`` sub-command
+~~~~~~~~~~~~~~~~~~~~~~
+
+Deletes an existing template. The options are:
```
[the remark this answers is not in the document]
Comment (Drab-Hounet) [reply documentation/en/source/pages/command-line/migration.rst]: Done but I wait @jfknoepfli 's opinion before to send a new patch

File: documentation/en/source/pages/command-line/migration.rst [hunk 4]
```diff
@@ -18,4 +18,14 @@ Sub Commands
...
+
+	* ``--id`` (mandatory): the ID of the template to delete
```
[the remark this answers is not in the document]
Comment (Drab-Hounet) [reply documentation/en/source/pages/command-line/migration.rst]: done

File: documentation/en/source/pages/command-line/migration.rst [hunk 5]
```diff
@@ -18,4 +18,14 @@ Sub Commands
...
+	* ``--no-confirm`` (optional): do not ask before delete the migration
```
[the remark this answers is not in the document]
Comment (Drab-Hounet) [reply documentation/en/source/pages/command-line/migration.rst]: done

File: hammr/commands/migration/migration.py [hunk 6]
```diff
@@ -17,10 +17,11 @@
 from ussclicore.cmd import Cmd, CoreGlobal
 from hammr.utils.hammr_utils import *
 from hammr.utils import migration_utils
+import shlex
 
 
 class Migration(Cmd, CoreGlobal):
-    """List existing migrations"""
+    """List existing migrations and delete an migration finished"""
```
[the remark this answers is not in the document]
Comment (Drab-Hounet) [reply hammr/commands/migration/migration.py]: done

File: hammr/commands/migration/migration.py [hunk 7]
```diff
@@ -53,3 +54,50 @@ def do_list(self, args):
     def help_list(self):
         doParser = self.arg_list()
         doParser.print_help()
+
+    def arg_delete(self):
+        doParser = ArgumentParser(prog=self.cmd_name + " delete", add_help=True,
+                                  description="Deletes an existing migration")
```
[the remark this answers is not in the document]
Comment (Drab-Hounet) [reply hammr/commands/migration/migration.py]: done

File: hammr/commands/migration/migration.py [hunk 8]
```diff
@@ -53,3 +54,50 @@ def do_list(self, args):
...
+        mandatory = doParser.add_argument_group("mandatory arguments")
+        mandatory.add_argument('--id', dest='id', required=True,
+                               help="the id of the migration to be deleted")
```
[the remark this answers is not in the document]
Comment (Drab-Hounet) [reply hammr/commands/migration/migration.py]: done

File: hammr/commands/migration/migration.py [hunk 9]
```diff
@@ -53,3 +54,50 @@ def do_list(self, args):
...
+        optional = doParser.add_argument_group("optional arguments")
+        optional.add_argument('--no-confirm', dest='no_confirm', action='store_true', required=False,
+                              help="do not ask before delete the migration")
```
[the remark this answers is not in the document]
Comment (Drab-Hounet) [reply hammr/commands/migration/migration.py]: done

File: tests/unit/commands/migration/test_migration.py (hunk 2 above)
Comment (Drab-Hounet) [reply tests/unit/commands/migration/test_migration.py]: Yes It's a good idea, I verify the execution in the two last tests.

File: tests/unit/commands/migration/test_migration.py (hunk 1 above)
Comment (Drab-Hounet) [reply tests/unit/commands/migration/test_migration.py]: done

File: tests/unit/commands/migration/test_migration.py [hunk 10]
```diff
@@ -71,6 +73,56 @@ def test_do_list_return_no_migration_message_when_there_is_no_migration(self, mo
...
+                                                                                    mock_migration_table):
+        # given
+        m, args = self.prepare_migrate_run_command()
+        mock_api_get.return_value = None
+
+        # when
+        m.do_delete(args)
+
+        # then
+        mock_migration_table(ANY).assert_not_called()
```
Comment (lqueiroga) [inline tests/unit/commands/migration/test_migration.py]: add call printer error no migration

File: tests/unit/commands/migration/test_migration.py [hunk 11]
```diff
@@ -71,6 +73,56 @@ def test_do_list_return_no_migration_message_when_there_is_no_migration(self, mo
...
+        migration = self.create_migration(1, "a migration", 50, "In Progress", False, False, False)
+        mock_api_get.return_value = migration
+        mock_query_yes_no.return_value = True
+
+        # when
+        m.do_delete(args)
+
+        # then
+        mock_migration_table.assert_called_with([migration])
+        mock_api_delete.assert_called_with()
```
Comment (lqueiroga) [inline tests/unit/commands/migration/test_migration.py]: add call printer ok migration deleted

Comment (Drab-Hounet) [reply tests/unit/commands/migration/test_migration.py]: done

File: tests/unit/commands/migration/test_migration.py (hunk 10 above)
Comment (Drab-Hounet) [reply tests/unit/commands/migration/test_migration.py]: done

File: documentation/en/source/pages/command-line/migration.rst [hunk 12]
```diff
@@ -18,4 +18,14 @@ Sub Commands
...
 
-Displays all the migrations for the user.
\ No newline at end of file
+Displays all the migrations for the user.
+
+
+``delete`` sub-command
+~~~~~~~~~~~~~~~~~~~~~~
+
+Deletes a complete migration. The options are:
```
Comment (cinlloc) [inline documentation/en/source/pages/command-line/migration.rst]: From my opinion, "complete migration" can be understood as "finished migration" or "full migration". Can be a little bit confusing.

File: documentation/en/source/pages/command-line/migration.rst [hunk 13]
```diff
@@ -18,4 +18,14 @@ Sub Commands
 ``list`` sub-command
 ~~~~~~~~~~~~~~~~~~~~
 
-Displays all the migrations for the user.
\ No newline at end of file
+Displays all the migrations for the user.
+
+
+``delete`` sub-command
```
Comment (cinlloc) [inline documentation/en/source/pages/command-line/migration.rst]: Should remove this not to conflict https://github.com/usharesoft/hammr/pull/357?

**Comment by lqueiroga** [comment]
Thanks for this PR

**Pull request merged**